cognite/client/utils/_auxiliary.py

```diff
@@ -65,6 +68,14 @@ def fast_dict_load(
     return instance
 
 
+def load_yaml_or_json(resource: str) -> dict[str, Any]:
+    try:
+        yaml = local_import("yaml")
```

Comment:
This will potentially get invoked a lot - is the overhead from importing negligible? If not, we can do a conditional definition of this method. e.g.

```
try:
  import yaml
  def load_yaml_or_json(resource: str) -> dict[str, Any]:
    return yaml.safe_load(resource)
except CogniteImportError:
  import json
  def load_yaml_or_json(resource: str) -> dict[str, Any]:
    return json.loads(resource)
  
```

Reply:
What about simply skipping the extra function call and do 
```
    try:
        import yaml
        return yaml.safe_load(resource)
    except ImportError:
        return json.loads(resource)
```